metricbeat/mb/mb.go

```diff
@@ -84,6 +90,42 @@ type EventsFetcher interface {
 	Fetch() ([]common.MapStr, error)
 }
 
+// Reporter is used by a MetricSet to report events, errors, or errors with
+// metadata. The methods return false iff publishing failed because the
+// MetricSet is being closed.
+type Reporter interface {
+	Event(event common.MapStr) bool
+	ErrorWith(err error, meta common.MapStr) bool
+	Error(error) bool
```

Comment:
Do we need both interfaces or could we just have `Error(err error, meta common.MapStr)` and in case the meta does not exist it is set to nil? Thinking if we could reduce the number of interface methods.